src/components/AddSongMenu.tsx

```diff
@@ -0,0 +1,226 @@
+  currentSong: string;
+};
+
+interface PlayListWithTracksInterface
+  extends SpotifyApi.PlaylistObjectSimplified {
+  songsOnPlayList: SpotifyApi.PlaylistTrackObject[];
+}
+
+export default function AddSongMenu({ currentSong }: AddMenuProps) {
+  const { spotifyApi } = useContext(AppContext);
+  const [playListWithTracks, setPlayListWithTracks] = useState<
+    PlayListWithTracksInterface[]
+  >([]);
+  const [state, setState] = useState({
+    bottom: false,
+  });
+  const [playListName, setPlayListName] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const showFetchingDataMessage = <CircularProgress />;
+
+  async function fetchPlayList() {
+    try {
+      setIsLoading(true);
+      const playListData = await spotifyApi.getUserPlaylists();
+      const playlistAndSongs = await Promise.all(
+        playListData.items.map(async (list) => {
+          const { items } = await spotifyApi.getPlaylistTracks(list.id);
+          const songsOnPlayList = items;
+          return { ...list, songsOnPlayList };
+        })
+      );
+
+      setPlayListWithTracks(playlistAndSongs);
+      setIsLoading(false);
+    } catch (error) {
+      handleExpiredToken(error);
+    }
+  }
+
+  const toggleDrawer = (anchor: string, open: boolean) => () => {
+    setState({ ...state, [anchor]: open });
+    if (open) {
+      fetchPlayList();
+    }
+  };
+
+  const list = (anchor: string) => {
+    return (
+      <Box
+        sx={{ width: anchor === "bottom" ? "auto" : 250 }}
+        role="presentation"
+      >
+        <List>
+          {playListWithTracks.map((list) => {
+            const included = list.songsOnPlayList.some(
+              (list) => list.track.uri === currentSong
+            );
+
+            return (
+              <ListItem
+                key={list.id}
+                disablePadding
+                onClick={() => handleListItemClick(list)}
+                onKeyDown={(event) => {
+                  if (event.key === "Enter") {
+                    handleListItemClick(list);
+                  }
+                }}
+              >
+                <ListItemButton>
+                  <ListItemAvatar>
+                    <Avatar
+                      alt={`{list.name}`}
+                      src={list.images !== null ? list.images[0].url : ""}
+                      variant="square"
+                    />
+                  </ListItemAvatar>
+
+                  <ListItemText primary={list.name} />
+                  {included && (
+                    <ListItemText primary="Song is already on list" />
+                  )}
+                </ListItemButton>
+              </ListItem>
+            );
+          })}
+        </List>
+      </Box>
+    );
+  };
+
+  async function handleListItemClick(
+    selectedPlayList: SpotifyApi.PlaylistObjectSimplified
+  ) {
+    try {
+      const song = await spotifyApi.getMyCurrentPlayingTrack();
+      const songToAdd = song?.item?.uri as string;
```

Comment:
`song` has the type `CurrentlyPlayingReponse` (not `null` or `undefined`) so the "?" is not needed here. (For `item` it is.)